R/messages.R

```diff
@@ -22,8 +22,9 @@ make_message <- function(id, title, message, db, queue, lockdir) {
   dir.create(lockdir, recursive = TRUE, showWarnings = FALSE)
   lock <- message_lock_file(lockdir, queue, id)
   con <- db_connect(lock)
-  db_execute(con, "CREATE TABLE foo (id INT)")
-  db_execute(con, "BEGIN EXCLUSIVE")
+  # Order is important here!
+  dbExecute(con, "CREATE TABLE foo (id INT)")
+  dbExecute(con, "BEGIN IMMEDIATE")
```

Comment:
I've tested current changes and it works great for my use case.

However, I'd like to ask why is this particular order important here. AFAIK, SQLite locks the whole database, not individual tables, and `CREATE TABLE` is a write operation. With this ordering, my test in #28 shows `FAILED` messages that are afterwards recovered. If I invert these two instructions, I don't see any failed messages anymore.

Reply:
Because SQLite will not lock a DB that has no tables, apparently, even with `BEGIN IMMEDIATE`. I am not entirely sure, but this is my best guess.

Did you see the `FAILED` messages with or without the recent changes in RSQLite? In your test `FAILED` means that (R)SQLite has given up and liteq needs to requeue?

Reply:
> Because SQLite will not lock a DB that has no tables, apparently, even with `BEGIN IMMEDIATE`. I am not entirely sure, but this is my best guess.

But this doesn't apply here. Whenever liteq is trying to set a new message, there's a queue set up, and thus there's at least one table. So, for messages, I believe, `BEGIN IMMEDIATE` should be before `CREATE TABLE`.

> Did you see the `FAILED` messages with or without the recent changes in RSQLite? In your test `FAILED` means that (R)SQLite has given up and liteq needs to requeue?

Without.

Reply:
> But this doesn't apply here.

It actually does. This is a different DB. That's why it is mostly managed differently from the `db.R` functions. This is an (almost) empty db, that we use as a lock file. (I wrote liteq before the filelock package.)

> Without

Can you please try with the latest RSQLite that has the changes? It is on CRAN now.

Reply:
Actually, it was with the latest RSQLite, I wasn't aware of the update: half of the messages fail with current ordering, no fails with `BEGIN IMMEDIATE` in the first place.

Reply:
Ok, I'll investigate...

Reply:
Actually, I have test failures and incorrect behavior with both versions. :(